### PR TITLE
Add refresh button for 0.7 skins

### PR DESCRIPTION
**Pull request by ChillerDragon**
See #8750

https://github.com/user-attachments/assets/a9cc0ea4-df3d-42b3-a69b-a83a9aaf534b

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/gameclient.cpp [hunk 1]
```diff
@@ -3682,6 +3682,14 @@ void CGameClient::RefreshSkins()
 		}
 	});
 
+	m_Skins7.Refresh([&](int) {
+		// if skin refreshing takes to long, swap to a loading screen
+		if(time_get_nanoseconds() - SkinStartLoadTime > 500ms)
+		{
+			m_Menus.RenderLoading(Localize("Loading 0.7 skin files"), "", 0, false);
```
Comment (Robyt3) [inline src/game/client/gameclient.cpp:3689]: `Loading skin files`. I don't think we need a different string here.

**Review by Robyt3 — CHANGES_REQUESTED** [review]

**Comment by ChillerDragon** [comment]
I just realized the chat skins are not refreshed. Converted to draft.

File: src/game/client/gameclient.cpp (hunk 1 above)
Comment (ChillerDragon) [reply src/game/client/gameclient.cpp:3689]: I thought it can help the user to know why the loading times are so slow. But sure I changed it back.

**Comment by ChillerDragon** [comment]
No idea how to get kill messages to update. They are only shown for a short time anyways. I do not want to spend more time on this. Everything else seems to work fine.

File: src/game/client/gameclient.h [hunk 2]
```diff
@@ -511,12 +511,7 @@ class CGameClient : public IGameClient
 	void OnInit() override;
 	void OnConsoleInit() override;
 	void OnStateChange(int NewState, int OldState) override;
-	template<typename T>
-	void ApplySkin7InfoFromGameMsg(const T *pMsg, int ClientId, int Conn);
-	void ApplySkin7InfoFromSnapObj(const protocol7::CNetObj_De_ClientInfo *pObj, int ClientId) override;
 	int OnDemoRecSnap7(class CSnapshot *pFrom, class CSnapshot *pTo, int Conn) override;
-	void *TranslateGameMsg(int *pMsgId, CUnpacker *pUnpacker, int Conn);
-	int TranslateSnap(CSnapshot *pSnapDstSix, CSnapshot *pSnapSrcSeven, int Conn, bool Dummy) override;
```
Comment (ChillerDragon) [inline src/game/client/gameclient.h:519]: In case anyone is wondering why I moved these. They were in the ``// hooks`` section which seems wrong.

**Review by def- — APPROVED** [review]

**Comment by Robyt3** [comment]
> No idea how to get kill messages to update. They are only shown for a short time anyways. I do not want to spend more time on this. Everything else seems to work fine.

Do they not get updated correctly by the existing `CInfoMessages::OnRefreshSkins`? To me it looks like this function should update the skins of all info messages correctly, including the sixup skin info, as it just copies the updated skin render info from the gameclient. The usage of `CTeeRenderInfo::Valid` for the info messages looks incorrect though. This function only checks if the 0.6 skin info is valid but it also needs to consider `CTeeRenderInfo::CSixup::Valid` I guess. We need to ensure all skin render info is updated, or the client may crash due to invalid texture handles being used, when a kill message uses a texture which gets unloaded because the respective skin was removed by refreshing.

**Comment by ChillerDragon** [comment]
> Do they not get updated correctly by the existing `CInfoMessages::OnRefreshSkins`? 

I removed the fade out code from the kill messages to have them stuck there so I can tested. Once I refresh the skins the old kill messages render no tee anymore and the new ones render the new tee.

> To me it looks like this function should update the skins of all info messages correctly, including the sixup skin info, as it just copies the updated skin render info from the gameclient.

Yeah thats what I thought. Not sure whats wrong here.

> The usage of `CTeeRenderInfo::Valid` for the info messages looks incorrect though. This function only checks if the 0.6 skin info is valid but we also need to consider `CTeeRenderInfo::CSixup::Valid` for the info message I guess.

Yeah good catch. But not sure how to apply this. I would like to avoid doing a bunch of `if(Client()->IsSixup ..)`` in the infomessages.cpp file. Also there is currently some hack in place that heinrich requested. Calling `RenderTee()` will pick a 0.7 or 0.6 renderer based on which render info is valid. So in 0.6 the 0.7 render info is expected to be invalid.

> We need to ensure all skin render info is updated, or the client may crash due to invalid texture handles being used, when a kill message uses a texture which gets unloaded because the respective skin was removed by refreshing.

Woah this sounds bad! Okay I did not know this. Converting back to draft because of that missing crucial crash fix.

**Comment by ChillerDragon** [comment]
I give up on this. Does not seem worth the effort to me since I have no idea how to fix the bug.

**Pull request closed without merging**